src/topology/task.rs

```diff
@@ -21,11 +23,43 @@ pub(crate) enum TaskOutput {
     Healthcheck,
 }
 
+#[derive(Debug, Snafu)]
+pub(crate) enum TaskError {
+    #[snafu(display("the task was cancelled before it completed"))]
+    Cancelled,
+    #[snafu(display("the task panicked and was aborted"))]
+    Panicked,
+    #[snafu(display("the task completed with an error"))]
+    Opaque,
+    #[snafu(display("{}", source))]
+    Wrapped {
+        source: Box<dyn error::Error + Send + Sync + 'static>,
```

Comment:
This (and below) is definitely `crate::Error`

Reply:
Yep!